docs/pages/l2_hdmi_cec_driver_test_specification.md

```diff
@@ -0,0 +1,221 @@
+|Description|After deleting the logical address, the test tries to send a broadcast command. It should fail to send during HAL Transmission call.|
+|Test Group|Module : 02|
+|Test Case ID|004|
+|Priority|High|
+
+**Pre-Conditions :** None
+
+**Dependencies :** None
+
+**User Interaction :**
+If user chose to run the test in interactive mode, then the test case has to be selected via console.
+
+#### Test Procedure :
+
+| Variation / Steps | Description | Test Data | Expected Result | Notes|
+| -- | --------- | ---------- | -------------- | ----- |
+| 01 | Open HDMI CEC with HdmiCecOpen | handle = valid handle | HDMI_CEC_IO_SUCCESS | Should be successful |
+| 02 | Set Rx Callback with HdmiCecSetRxCallback | handle = valid handle, callback = NULL, user_data = NULL | HDMI_CEC_IO_SUCCESS | Should be successful |
+| 03 | Add Logical Address with HdmiCecAddLogicalAddress | handle = valid handle, logicalAddresses = 0x0 | HDMI_CEC_IO_SUCCESS | Should be successful |
+| 04 | Remove Logical Address with HdmiCecRemoveLogicalAddress | handle = valid handle, logicalAddresses = 0x0 | HDMI_CEC_IO_SUCCESS | Should be successful |
+| 05 | Transmit with HdmiCecTx after removing logical address | handle = valid handle, buf = {0x3F, CEC_GET_CEC_VERSION}, len = 1, result = valid result buffer | return status != HDMI_CEC_IO_SUCCESS | Should be successful |
+| 06 | Close HDMI CEC with HdmiCecClose | handle = valid handle | HDMI_CEC_IO_SUCCESS | Should be successful |
+
+
+```mermaid
+graph TD
+  A[HdmiCecOpen API Call]
+  A -->|Success| B[HdmiCecSetRxCallback API Call]
+  A -->|Failure| A1[Test case fail]
+  B -->|Success| C[HdmiCecAddLogicalAddress API Call]
+  B -->|Failure| B1[Test case fail]
+  C -->|Success| D[HdmiCecRemoveLogicalAddress API Call]
+  C -->|Failure| C1[Test case fail]
+  D -->|Success| E[HdmiCecTx API Call]
+  D -->|Failure| D1[Test case fail]
+  E -->|Expected Failure| G[Test case pass] 
+  E -->|Unexpected Success| E1[Test case fail]
+  G --> H[HdmiCecClose API Call]
+  E1 --> H 
+```
+
+
+### Test 5
+
+|Title|Details|
+|--|--|
+|Function Name|`test_l2_hdmi_cec_driver_VerifyPhysicalAddressAllocation`|
+|Description|Verify the valid physical address allocated through the HAL function.|
+|Test Group|Module : 02|
+|Test Case ID|005|
+|Priority|High|
+
+**Pre-Conditions :** None
+
+**Dependencies :** None
+
+**User Interaction :** If user chose to run the test in interactive mode, then the test case has to be selected via console.
+
+#### Test Procedure :
+
+| Variation / Steps | Description | Test Data | Expected Result | Notes|
+| -- | --------- | ---------- | -------------- | ----- |
+| 01 | Open the HDMI CEC driver using HdmiCecOpen | handle = valid buffer | HDMI_CEC_IO_SUCCESS | Should be successful |
+| 02 | Get the physical address using HdmiCecGetPhysicalAddress | handle = valid handle, physicalAddress = valid buffer | HDMI_CEC_IO_SUCCESS | Should be successful |
```

Comment:
we need to check the valid address based on the data provided in the hal interface file.

Reply:
done
All input, output, pre-requisite, post-requisite comments are addressed in the upgraded GAI BOT